spinn_utilities/data/run_status.py

```diff
@@ -0,0 +1,36 @@
+# Copyright (c) 2021-2022 The University of Manchester
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from enum import Enum
+
+
+class RunStatus(Enum):
+    """
+    Different states the run could be in
+
+    This class is design to used internally by UtilsDataView
+    """
+    NOT_SETUP = (0)
+    NOT_RUNNING = (12)
+    IN_RUN = (2)
+    STOP_REQUESTED = (3)
+    STOPPING = (4)
+    SHUTDOWN = (5)
```

Comment:
This is from the prespective of the users script.
It says nothing about if there is c code running.
Ie Has a sim.run (or similar) call started but not yet returned.

Reply:
This is combined with the ResetStatus to get the needed status.

Reply:
"pause" as between two auto pause loops or between load executable and application runner not interesting from a DataView prospective.

"pause" as between one sim.run a later sim.run, sim.end ect is handled by a combination of reset_status and run_status.  
This split was done because of the number of different combination needed especially as the reset information is still needed while "In_RUN"